internal/server/routes.go

```diff
@@ -234,14 +235,23 @@ func wrapRoute[Req, Res any](a *API, routeID routeIdentifier, route route[Req, R
 			return err
 		}
 
-		if err := tx.Commit(); err != nil {
+		completeTx := tx.Commit
+		if route.txnOptions != nil && route.txnOptions.ReadOnly {
+			// use rollback to avoid an error when the request handler already completed the txn
```

Comment:
In what case would the transaction already be committed here?

Reply:
The transaction is not necessarily committed, but it should be completed (rollback or commit).

The one use case we have now is the one you commented on above. Long running transactions are bad for database performance (it consumes resources to keep the transaction open), so we need to end the transaction before blocking.

I should probably make a similar change for the `debug/pprof` endpoint, so that we don't keep a transaction open while profiling and tracing.